packages/firestore/lite/src/api/query.ts

```diff
@@ -0,0 +1,1247 @@
+ * offline, the operation fails. To see previously cached result and local
+ * modifications, use the full Firestore SDK.
+ *
+ * @param query - The `Query` to execute.
+ * @returns A Promise that will be resolved with the results of the query.
+ */
+export function getDocs<T>(query: Query<T>): Promise<QuerySnapshot<T>> {
+  query = cast<Query<T>>(query, Query);
+  validateHasExplicitOrderByForLimitToLast(query._query);
+
+  const datastore = getDatastore(query.firestore);
+  const userDataWriter = new LiteUserDataWriter(query.firestore);
+  return invokeRunQueryRpc(datastore, query._query).then(result => {
+    const docs = result.map(
+      doc =>
+        new QueryDocumentSnapshot<T>(
+          query.firestore,
+          userDataWriter,
+          doc.key,
+          doc,
+          query._converter
+        )
+    );
+
+    if (hasLimitToLast(query._query)) {
+      // Limit to last queries reverse the orderBy constraint that was
+      // specified by the user. As such, we need to reverse the order of the
+      // results to return the documents in the expected order.
+      docs.reverse();
+    }
+
+    return new QuerySnapshot<T>(query, docs);
+  });
+}
+
+/**
+ * Writes to the document referred to by the specified `DocumentReference`. If
+ * the document does not yet exist, it will be created.
+ *
+ * The result of this write will only be reflected in document reads that occur
+ * after the returned Promise resolves. If the client is offline, the
+ * write fails. If you would like to see local modifications or buffer writes
+ * until the client is online, use the full Firestore SDK.
+ *
+ * @param reference - A reference to the document to write.
+ * @param data - A map of the fields and values for the document.
+ * @returns A Promise resolved once the data has been successfully written
+ * to the backend.
+ */
+export function setDoc<T>(
+  reference: DocumentReference<T>,
+  data: T
+): Promise<void>;
+/**
+ * Writes to the document referred to by the specified `DocumentReference`. If
+ * the document does not yet exist, it will be created. If you provide `merge`
+ * or `mergeFields`, the provided data can be merged into an existing document.
+ *
+ * The result of this write will only be reflected in document reads that occur
+ * after the returned Promise resolves. If the client is offline, the
+ * write fails. If you would like to see local modifications or buffer writes
+ * until the client is online, use the full Firestore SDK.
+ *
+ * @param reference - A reference to the document to write.
+ * @param data - A map of the fields and values for the document.
+ * @param options - An object to configure the set behavior.
+ * @returns A Promise resolved once the data has been successfully written
+ * to the backend.
+ */
+export function setDoc<T>(
+  reference: DocumentReference<T>,
+  data: Partial<T>,
+  options: SetOptions
+): Promise<void>;
+
+export function setDoc<T>(
```

Comment:
Setting, updating, and adding documents doesn't seem to have anything to do with "queries". Did you mean to move this here given that `reference.ts` still exists?

Alternatively, if this can't exist in `reference.ts` to solve circularity issues, maybe create `reference_impl.ts` to match the structure you've established elsewhere (e.g. with LocalStoreImpl)?

Reply:
Sorry, this was also cleaned up in a later commit, which I know is impossible to detect with a PR this size.

 I did rename the file to reference_impl.ts as well.